sysdig/resource_sysdig_secure_notification_channel_msteams.go

```diff
@@ -0,0 +1,213 @@
+package sysdig
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	v2 "github.com/draios/terraform-provider-sysdig/sysdig/internal/client/v2"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceSysdigSecureNotificationChannelMSTeams() *schema.Resource {
+	timeout := 5 * time.Minute
+
+	return &schema.Resource{
+		CreateContext: resourceSysdigSecureNotificationChannelMSTeamsCreate,
+		UpdateContext: resourceSysdigSecureNotificationChannelMSTeamsUpdate,
+		ReadContext:   resourceSysdigSecureNotificationChannelMSTeamsRead,
+		DeleteContext: resourceSysdigSecureNotificationChannelMSTeamsDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(timeout),
+			Update: schema.DefaultTimeout(timeout),
+			Read:   schema.DefaultTimeout(timeout),
+			Delete: schema.DefaultTimeout(timeout),
+		},
+
+		Schema: createSecureNotificationChannelSchema(map[string]*schema.Schema{
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"template_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		}),
+	}
+}
+
+func resourceSysdigSecureNotificationChannelMSTeamsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := getSecureNotificationChannelClient(meta.(SysdigClients))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	teamID, err := client.CurrentTeamID(ctx)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	notificationChannel, err := secureNotificationChannelMSTeamsFromResourceData(d, teamID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	notificationChannel, err = client.CreateNotificationChannel(ctx, notificationChannel)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(strconv.Itoa(notificationChannel.ID))
```

Comment:
Instaed of doing this d.Set call `resourceSysdigSecureNotificationChannelMSTeamsRead` which will populate everything.